master/internal/command/command.go

```diff
@@ -146,357 +139,326 @@ func remakeCommandsByType(
-		}
-		if err := user.DeleteSessionByToken(
-			context.TODO(),
-			c.GenericCommandSpec.Base.UserSessionToken,
-		); err != nil {
-			ctx.Log().WithError(err).Errorf(
-				"failure to delete user session for task: %v", c.taskID)
-		}
-		actors.NotifyAfter(ctx, terminatedDuration, terminateForGC{})
-	case getSummary:
-		if msg.userFilter == "" || c.Base.Owner.Username == msg.userFilter {
-			ctx.Respond(c.summary(ctx))
-		}
+	if err := c.snapshotAndPersist(ctx); err != nil {
+		c.syslog.WithError(err).Warnf("command persist failure")
+	}
+	return nil
+}
 
-	case *notebookv1.Notebook:
-		ctx.Respond(c.toNotebook(ctx))
-
-	case *apiv1.GetNotebookRequest:
-		ctx.Respond(&apiv1.GetNotebookResponse{
-			Notebook: c.toNotebook(ctx),
-			Config:   protoutils.ToStruct(c.Config),
-		})
-	case *apiv1.KillNotebookRequest:
-		// TODO(Brad): Do the same thing to allocations that we are doing to RMs.
-		err := task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
-		if err != nil {
-			ctx.Log().WithError(err).Warn("failed to kill allocation")
-		}
-		ctx.Respond(&apiv1.KillNotebookResponse{Notebook: c.toNotebook(ctx)})
-	case *apiv1.SetNotebookPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
-		}
-		ctx.Respond(&apiv1.SetNotebookPriorityResponse{Notebook: c.toNotebook(ctx)})
+// registerJobAndTask registers the command with the job service & adds the command to the job & task dbs.
+func (c *command) registerJobAndTask(ctx context.Context, tx bun.Tx) error {
+	c.registeredTime = time.Now()
+	// TODO CAROLINA: put this in a tx
+	if err := db.AddJobTx(ctx, tx, &model.Job{
+		JobID:   c.jobID,
+		JobType: c.jobType,
+		OwnerID: &c.Base.Owner.ID,
+	}); err != nil {
+		return errors.Wrapf(err, "persisting job %v", c.taskID)
+	}
 
-	case *commandv1.Command:
-		ctx.Respond(c.toCommand(ctx))
+	if err := db.AddTaskTx(ctx, tx, &model.Task{
+		TaskID:     c.taskID,
+		TaskType:   c.taskType,
+		StartTime:  c.registeredTime,
+		JobID:      &c.jobID,
+		LogVersion: model.CurrentTaskLogVersion,
+	}); err != nil {
+		return errors.Wrapf(err, "persisting task %v", c.taskID)
+	}
 
-	case *apiv1.GetCommandRequest:
-		ctx.Respond(&apiv1.GetCommandResponse{
-			Command: c.toCommand(ctx),
-			Config:  protoutils.ToStruct(c.Config),
-		})
+	if c.contextDirectory == nil {
+		c.contextDirectory = make([]byte, 0)
+	}
 
-	case *apiv1.KillCommandRequest:
-		err := task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
-		if err != nil {
-			ctx.Log().WithError(err).Warn("failed to kill allocation")
-		}
-		ctx.Respond(&apiv1.KillCommandResponse{Command: c.toCommand(ctx)})
+	if _, err := tx.NewInsert().Model(&model.TaskContextDirectory{
+		TaskID:           c.taskID,
+		ContextDirectory: c.contextDirectory,
+	}).Exec(ctx); err != nil {
+		return fmt.Errorf("persisting context directory files: %w", err)
+	}
+	c.contextDirectory = nil
```

Comment:
i liked the old function `c.persistAndEvictContextDirectoryFromMemory()`. made it very clear by name why this last, subtle line was so important. can we bring it back and just pass it a `tx`?